doc/arch/adr-005-incremental-popularity-updates.md

```diff
@@ -0,0 +1,68 @@
+# Decision record: Incremental popularity updates
+
+Date: 2017-09-15
+
+## Status
+
+Rejected.
+
+## Context
+
+In [ADR 003 - perform popularity updating without using an index lock](doc/arch/adr-003-popularity-updating-without-index-locks.md) we changed the process
+for bulk loading data into Rummager's search indexes. The main reason we bulk load
+data is to update the popularity field every night, which affects every document in the search index.
+
+Previously we'd used a separate search index to write into, and locked the existing index
+for writes until the process completed. Then we'd switch the read alias to the new index.
+
+We decided to change this to update the existing index in place, which removed
+the need for locking. Although the task usually runs overnight, we wanted to minimise the time we locked the index for, so we could run it in the day if necessary. We used sidekiq to distribute bulk index requests over multiple workers.
+
+## Problem
+
+After switching to the new index, we observed regular spikes in 503 errors from the search API, that coincide with when the nightly task runs. The elasticsearch queries were taking longer than normal and timing out.
+
+We want reads from the index to be unaffected by bulk indexing, because the user's ability to search is more important than the content being up to date.
+
+![Spike in slow requests](images/popularity-errors.png)
+
+In light of this, we considered some ways we could change the implementation of the nightly update process to improve performance.
+
+## Proposal: Update less data
+
+The `page-traffic` index contains today's analytics data, and is updated nightly. The number of page views a link has over a 14 day period is stored in its `vc_14` field, and the rank relative to other pages on GOV.UK is stored in the `rank_14` field.
+
+We considered limiting the popularity update task so that it only changed
+a small number of documents. We compared the `vc_14` field in the page traffic
+over two successive days to work out how much page popularity shifts in practice.
+
+A very small number number of pages actually have significant changes day to day
+(i.e, a change of at least +/- 10 page views). This means that we shouldn't need
+to update the entire search index every day - we could get away with 10% of the current update workload.
+
+![Changes in VC-14 field day to day](images/changes-in-vc-14-metric.png)
+
+## Decision
+
+We decided against making any technical changes to the popularity update process now. Instead we're going to reduce the number of sidekiq processes that index into
+elasticsearch at the same time, which reduces the impact on search performance.
+
+We don't think that the incremental popularity updates would work
```

Comment:
This feels more like "learnings" than "decision". And hide the fact the first paragraph here gives the information on why this ADR was rejected and what it was rejected in favour of.